fastly/ngwaf/v1/events/api_expire.go

```diff
@@ -0,0 +1,57 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/fastly/go-fastly/v10/fastly"
+)
+
+// ExpireInput specifies the information needed for the Expire() function to
+// perform the operation.
+type ExpireInput struct {
+	// Context, if supplied, will be used as the Request's context.
+	Context *context.Context `json:"-"`
+	// EventID is the event identifier (required).
+	EventID *string `json:"-"`
+	// IsExpired sets the value of IsExpired on the event (required).
+	IsExpired *bool `json:"is_expired"`
+	// WorkspaceID is the workspace identifier (required).
+	WorkspaceID *string `json:"-"`
+}
+
+// Expire expires the specified event.
+func Expire(c *fastly.Client, i *ExpireInput) (*Event, error) {
+	if i.WorkspaceID == nil {
+		return nil, fastly.ErrMissingWorkspaceID
+	}
+	if i.EventID == nil {
+		return nil, fastly.ErrMissingEventID
+	}
+	if i.IsExpired == nil {
+		return nil, fastly.ErrMissingIsExpired
+	}
+
+	requestOptions := fastly.CreateRequestOptions(i.Context)
+
+	bodyString := map[string]bool{"is_expired": *i.IsExpired}
+	marshalledBody, _ := json.Marshal(bodyString)
```

Comment:
Is there a benefit to this method over just marshaling the ExpireInput structure itself?